src/versionbits.cpp

```diff
@@ -105,6 +104,42 @@ ThresholdState AbstractThresholdConditionChecker::GetStateFor(const CBlockIndex*
     return state;
 }
 
+// return the numerical statistics of blocks signalling the specified BIP9 condition in this current period
+BIP9Stats AbstractThresholdConditionChecker::GetStateStatisticsFor(const CBlockIndex* pindexPrev, const Consensus::Params& params) const
+{
+    BIP9Stats obj;
+    
+    int elapsedBlocks;
+    int nPeriod = Period(params);
+    int nThreshold = Threshold(params);
+    
+    const CBlockIndex* currentIndex = pindexPrev;
+    
+    obj.period = nPeriod;
+    obj.threshold = nThreshold;
+
+    // Find beginning of period
+    if (pindexPrev != NULL) {
```

Comment:
Moved the check up so the return struct is partially initialized. However, this function shouldn't ever be called unless `THRESHOLD_STARTED == thresholdState` anyway, which will never be the case at block 0.